web/client/codechecker_client/helpers/base.py

```diff
@@ -28,13 +32,43 @@ def __init__(self, protocol, host, port, uri, session_token=None,
         self.__port = port
         url = create_product_url(protocol, host, port, uri)
 
-        self.transport = THttpClient.THttpClient(url)
+        self.transport = None
+
+        try:
+            self.transport = THttpClient.THttpClient(url)
+        except ValueError:
+            # Initalizing THttpClient may raise an exception if proxy settings
+            # are used but the port number is not a valid integer.
+            pass
+        finally:
+            # Thrift do not handle the use case when invalid proxy format is
+            # used (e.g.: no schema is specified). For this reason we need to
+            # verify the proxy format in our side.
+            self._validate_proxy_format()
```

Comment:
If THttpClient throws TTransportException for some reason but not because something bad with proxy settings then this finally branch try to validate proxy settings.
Please call self._validate_proxy_format() in the ValueError branch not in the finally branch.

Reply:
Only the `THttpClient` object creation is inside a try/except block, which cannot raise such exception like `TTransportException`.
Two scenarios are possible here:
- The user specified the following environment variable: `HTTP_PROXY=http://localhost` / `HTTP_PROXY=http://localhost:not_a_number` where the port is not specified or it is not an integer. In this case it will raise a ValueError exception.
- The user specified the following environment variable: `HTTP_PROXY=localhost:8001` where the schema is not given. In this case the `THttpClient` object creation will not raise any exception (it will raise exception when an API call happens). So this is the reason when this object creation is done, we need to check the proxy format before any API call happens.

In both cases we need to check the proxy format. So it is not enough to call this function inside the ValueError branch. We need to call it every time when there is no exception.